src/keystore.cpp

```diff
@@ -11,6 +11,22 @@ bool CKeyStore::AddKey(const CKey &key) {
     return AddKeyPubKey(key, key.GetPubKey());
 }
 
+void CBasicKeyStore::ImplicitlyLearnRelatedKeyScripts(const CPubKey& pubkey)
+{
+    AssertLockHeld(cs_KeyStore);
+    // This adds the redeemscripts necessary to detect P2WPKH and P2SH-P2WPKH
+    // outputs. Technically P2WPKH outputs don't have a redeemscript to be
+    // spent. However, our current IsMine logic requires the corresponding
+    // P2SH-P2WPKH redeemscript to be present in the wallet in order to accept
+    // payment even to P2WPKH outputs.
```

Comment:
In commit "Implicitly know about P2WPKH redeemscripts"

Maybe add a sentence like:

> "Implicitly" refers to fact that scripts are derived automatically from existing keys, and are only stored in memory, not written to disk.

It might also be worth saying why they aren't written to disk. It's not clear if that would be completely useless in any downgrade scenario or if it just doesn't help with one of the downgrade scenarios we care about.

Reply:
Adding.